mptcp:redundant fix rtx-tree migration bug

vuva opened the pull request:
After the commit introducing rt-tree based retransmit
queue, skb(s) that have been transmitted by a subflow are moved to rt-tree
rtx_queue and deleted from sk_write_queue. If the subflows have different
sending rate, the slower subflow will miss those skb(s) if it doesn't check the
rtx_queue.

This fix makes sure the Redundant scheduler checks the rt-tree based
rtx_queue when it selects the skb to schedule, and check
tcp_rtx_queue_empty before it selects the next segment.

Fixes: c61bc63e8f37 ("Merge tag 'v4.15-rc3' into mptcp_trunk")
Closes: https://github.com/multipath-tcp/mptcp/issues/392
Signed-off-by: Vu Vu <vuanh.vu@ikt.uni-hannover.de>

vuva committed:
fix redundant scheduler rt-tree migration bug

matttbe committed:
to-squash: apply kernel code style

matttbe committed:
to-squash: code style + avoid two calls to func

one line was removed in the first commit separating the variables
declaration and the rest.

also, no need to call tcp_rtx_queue_head() twice.

matttbe commented:
Hi @vuva 

Thank you for the patch!

It is now applied (after having applied the kernel code style and fix checkpatch errors) in mptcp_trunk and backported to mptcp_v0.95.

Cheers,
Matt

matttbe commented:
Hi @vuva ,
When reading Christoph's release notes of the v0.95.1 version, I noticed I accidentally changed the author when I squashed the 2 commits I added, sorry, my bad, that was not my intension! I guess the `git merge --squash` I used set myself as the author instead of re-using the author of the first commit.

vuva commented:
> 
> 
> Hi @vuva ,
> When reading Christoph's release notes of the v0.95.1 version, I noticed I accidentally changed the author when I squashed the 2 commits I added, sorry, my bad, that was not my intension! I guess the `git merge --squash` I used set myself as the author instead of re-using the author of the first commit.

No problem, @matttbe . It doesn't matter. The fix was merged anyway.
